src/Sulu/Bundle/TagBundle/Tag/TagManager.php

```diff
@@ -53,16 +58,23 @@ class TagManager implements TagManagerInterface
      */
     private $domainEventCollector;
 
+    /**
+     * @var TrashManagerInterface
+     */
+    private $trashManager;
+
     public function __construct(
         TagRepositoryInterface $tagRepository,
         ObjectManager $em,
         EventDispatcherInterface $eventDispatcher,
-        DomainEventCollectorInterface $domainEventCollector
+        DomainEventCollectorInterface $domainEventCollector,
+        TrashManagerInterface $trashManager
```

Comment:
Would rather not do this, since all future constructor arguments after `$trashManager` also need `= null` then

Reply:
`?TrashManagerInterface $trashManager` should do the job too

Reply:
The future constructor arguments should be `null` and trigger a deprecation or correct error message from BC promise. Adding arguments to service constructors make the container hard fail so this should be if possible always be avoided to avoid errors when a symfony application is deployed without green blue deployment.

Reply:
So I definitely would add ` = null` as it does not hurt and the __construct doesn't care about inheritance matching.